utils/python/CIME/test_status.py

```diff
@@ -205,11 +225,8 @@ def get_overall_test_status(self, wait_for_run=False, check_throughput=False, ch
         'PASS'
         """
         rv = TEST_PASS_STATUS
-        run_phase_found = False
         for phase, data in self._phase_statuses.iteritems():
             status = data[0]
-            if phase == RUN_PHASE:
```

Comment:
Since this "run_phase_found" logic is now needed in two places, I refactored it to a _has_run_phase method